manage.php

```diff
@@ -65,44 +74,26 @@
 if (optional_param('remove', false, PARAM_BOOL) && confirm_sesskey()) {
     $userstounassign = $currentuserselector->get_selected_users();
     if (!empty($userstounassign)) {
-        foreach($userstounassign as $removeuser) {
-            \availability_shibboleth2fa\condition::set_exception($course->id, $removeuser->id, false);
+        foreach ($userstounassign as $removeuser) {
+            condition::set_exception(
+                courseid: $course->id,
+                userid: $removeuser->id,
+                skipauth: false,
+            );
         }
-
         $potentialuserselector->invalidate_selected_users();
         $currentuserselector->invalidate_selected_users();
     }
 }
 
 echo $OUTPUT->heading(get_string('user_exceptions', 'availability_shibboleth2fa'));
-
-?>
-<form id="assignform" method="post" action="<?php echo $PAGE->url ?>"><div>
-  <input type="hidden" name="sesskey" value="<?php echo sesskey() ?>" />
-
-  <table summary="" class="roleassigntable generaltable generalbox boxaligncenter" cellspacing="0">
-    <tr>
-      <td id="existingcell">
-          <p><label for="removeselect"><?php print_string('users_with_exception', 'availability_shibboleth2fa'); ?></label></p>
-          <?php $currentuserselector->display() ?>
-      </td>
-      <td id="buttonscell">
-          <div id="addcontrols">
-              <input name="add" id="add" type="submit" value="<?php echo $OUTPUT->larrow().'&nbsp;'.get_string('add'); ?>" title="<?php print_string('add'); ?>" /><br />
-          </div>
-
-          <div id="removecontrols">
-              <input name="remove" id="remove" type="submit" value="<?php echo get_string('remove').'&nbsp;'.$OUTPUT->rarrow(); ?>" title="<?php print_string('remove'); ?>" />
-          </div>
-      </td>
-      <td id="potentialcell">
-          <p><label for="addselect"><?php print_string('users_without_exception', 'availability_shibboleth2fa'); ?></label></p>
-          <?php $potentialuserselector->display() ?>
-      </td>
-    </tr>
-  </table>
-</div></form>
-<?php
-
-
+$templatecontext = [
+    'actionurl' => $PAGE->url->out(),
```

Comment:
I think the URL is escaped twice: Once by the `out()` function and once in the template.